pmd-java/src/main/resources/category/java/codestyle.xml

```diff
@@ -484,12 +484,15 @@ usage by developers who should be implementing their own versions in the concret
                 <value>
 <![CDATA[
 //ClassDeclaration[@RegularClass = true() and pmd-java:modifiers() = "abstract"]
-    /ClassBody
+  /ClassBody
     /MethodDeclaration
+    [not(@Name = "finalize" and @Final = true())]
```

Comment:
Sad that you restrict it to `finalize`.
Any method would be fine where you want subclasses prevent overriding them.

Btw. you forgot the condition that the class is non-final.
In a final class, the method again becomes pointless and should be removed,
for example if you previously did it on method level and later make the whole class final.